### PR TITLE
crosscluster, upgrade: bootstrap migration to setup reader catalog an…

### LINKED ISSUE #130445: implement reader tenant job
Create a reader tenant job on tenant startup.

Epic: CRDB-23575

Jira issue: CRDB-42055

### PR DESCRIPTION
**Pull request by azhu-crl**
…d job

This commit creates a `StandbyReadTSPoller` job on tenant startup if read from standby has been enabled.

Fixes: #130445
Epic: CRDB-23575
Release note: None

### CODE REVIEW

File: pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	}
+	id, ts, err := d.TenantInfoAccessor.ReadFromTenantInfo(ctx)
+	if err != nil {
+		return err
+	}
+	if !id.IsSet() {
+		return nil
+	}
+	if ts.IsEmpty() {
+		log.Warning(ctx, "replicated timestamp is zero")
```
Comment (dt) [inline pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go:39]: `return nil` ?

File: pkg/ccl/crosscluster/physical/standby_read_ts_poller_job_test.go [hunk 2]
```diff
@@ -58,17 +56,15 @@ func TestStandbyReadTSPollerJob(t *testing.T) {
...
 	registry := c.ReaderTenantServer.JobRegistry().(*jobs.Registry)
-	jobRecord := makeStandbyReadTSPollerJobRecord(registry, username.MakeSQLUsernameFromPreNormalizedString("user"))
-	sqlDB := c.ReaderTenantServer.InternalDB().(isql.DB)
 
-	err := sqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
-		_, err := registry.CreateJobWithTxn(ctx, jobRecord, jobRecord.JobID, txn)
-		return err
+	testutils.SucceedsSoon(t, func() error {
+		runningJobs := registry.CurrentlyRunningJobs()
+		if len(runningJobs) != 1 {
```
Comment (dt) [inline pkg/ccl/crosscluster/physical/standby_read_ts_poller_job_test.go:62]: there are lots of other jobs in this registry aren't there?

Comment (dt) [reply pkg/ccl/crosscluster/physical/standby_read_ts_poller_job_test.go:62]: I think I'd go back to `WaitForJobToRun`, and just change it to find the job ID based on querying for the job by job type in `crdb_internal.jobs`

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
@azhu-crl the extended CI failure looks fun 😁

File: pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go (hunk 1 above)
Comment (msbutler) [reply pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go:39]: we need to set up the poller job even if the timestamp is 0

Comment (msbutler) [reply pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go:39]: if the ts is 0, we just need to avoid calling `SetupOrAdvanceStandbyReaderCatalog`

**Review by msbutler — CHANGES_REQUESTED** [review]
Looks good, but i think you need to tweak the logic around an empty replicated time.

File: pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go (hunk 1 above)
Comment (dt) [reply pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go:39]: But only creating the poller works too and is arguably the more defensive approach

Comment (azhu-crl) [reply pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go:39]: I updated the code to create the poller job only if replicated time is zero

**Review by msbutler — APPROVED** [review]

**Comment by msbutler** [comment]
Thanks!

**Comment by azhu-crl** [comment]
TFTR!

bors r=msbutler,dt

**Pull request merged**